code/android-core-library/src/main/java/com/adobe/marketing/mobile/configuration/ConfigurationExtension.kt

```diff
@@ -0,0 +1,454 @@
+import com.adobe.marketing.mobile.ExtensionEventListener
+import com.adobe.marketing.mobile.LoggingMode
+import com.adobe.marketing.mobile.MobileCore
+import com.adobe.marketing.mobile.internal.compatibility.CacheManager
+import com.adobe.marketing.mobile.launch.rulesengine.LaunchRulesEngine
+import com.adobe.marketing.mobile.launch.rulesengine.LaunchRulesEvaluator
+import com.adobe.marketing.mobile.services.CacheFileService
+import com.adobe.marketing.mobile.services.ServiceProvider
+import java.lang.Exception
+import java.util.concurrent.Executors
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+
+/**
+ * Configuration Extension.
+ * Responsible for retrieving the configuration of the SDK, updating the shared state and
+ * dispatching configuration updates through the `EventHub`
+ */
+internal class ConfigurationExtension : Extension {
+
+    companion object {
+        private const val TAG = "ConfigurationExtension"
+        private const val EXTENSION_NAME = "com.adobe.module.configuration"
+        private const val EVENT_TYPE = "com.adobe.eventType.configuration"
+        private const val EVENT_SOURCE_REQUEST_CONTENT = "com.adobe.eventSource.requestContent"
+        private const val EVENT_SOURCE_RESPONSE_CONTENT = "com.adobe.eventSource.responseContent"
+
+        internal const val CONFIGURATION_REQUEST_CONTENT_JSON_APP_ID = "config.appId"
+        internal const val CONFIGURATION_REQUEST_CONTENT_JSON_FILE_PATH = "config.filePath"
+        internal const val CONFIGURATION_REQUEST_CONTENT_JSON_ASSET_FILE = "config.assetFile"
+        internal const val CONFIGURATION_REQUEST_CONTENT_UPDATE_CONFIG = "config.update"
+        internal const val CONFIGURATION_REQUEST_CONTENT_CLEAR_UPDATED_CONFIG =
+            "config.clearUpdates"
+        internal const val CONFIGURATION_REQUEST_CONTENT_RETRIEVE_CONFIG = "config.getData"
+        internal const val CONFIGURATION_REQUEST_CONTENT_IS_INTERNAL_EVENT =
+            "config.isinternalevent"
+
+        internal const val DATASTORE_KEY = "AdobeMobile_ConfigState"
+        internal const val RULES_CONFIG_URL = "rules.url"
+
+        internal const val CONFIG_DOWNLOAD_RETRY_ATTEMPT_DELAY_MS = 5L
+        internal const val CONFIGURATION_RESPONSE_IDENTITY_ALL_IDENTIFIERS = "config.allIdentifiers"
+        internal const val EVENT_STATE_OWNER = "stateowner"
+        internal const val GLOBAL_CONFIG_PRIVACY = "global.privacy"
+    }
+
+    private val serviceProvider: ServiceProvider
+    private val appIdManager: AppIdManager
+    private val cacheFileService: CacheFileService
+    private val launchRulesEvaluator: LaunchRulesEvaluator
+    private val configurationStateManager: ConfigurationStateManager
+    private val configurationRulesManager: ConfigurationRulesManager
+    private val retryWorker: ScheduledExecutorService
+    private val extensionEventListener: ExtensionEventListener = ExtensionEventListener {
```

Comment:
Nit: Do you think we can pass it inline?

Reply:
+1. 
I would prefer avoiding this variable and passing it as lambda to `registerEventListener` API.